application/client/src/app/ui/elements/timezones/timezone.ts

```diff
@@ -30,38 +34,23 @@ export class Timezone {
         );
     }
 
-    public getNameAsHtml(): string {
-        if (this._filter === '') {
-            return this.name;
-        }
-        const reg = regex.fromStr(this._filter);
-        if (reg instanceof Error) {
-            return this.name;
-        }
-        return this.name.replace(reg, (match): string => {
-            return `<span>${match}</span>`;
-        });
+    public get html_name(): string {
+        return this._html_name;
     }
 
-    public getUtcAsHtml(): string {
-        if (this._filter === '') {
-            return this.utc;
-        }
-        const reg = regex.fromStr(this._filter);
-        if (reg instanceof Error) {
-            return this.utc;
-        }
-        return this.utc.replace(reg, (match): string => {
-            return `<span>${match}</span>`;
-        });
+    public get html_utc(): string {
+        return this._html_utc;
     }
 
     public filter(filter: string) {
-        this._filter = filter.trim();
-        this.hidden =
-            this._filter === ''
-                ? false
-                : this.name.toLowerCase().indexOf(filter) === -1 &&
-                  this.utc.toLowerCase().indexOf(filter) === -1;
+        let name = this._matcher.search_single(filter, this.name);
+        let utc = this._matcher.search_single(filter, this.utc);
```

Comment:
That's okay. I mean `search_single`, because this methods comes from rust... it can be out of a JS naming convention